### PR TITLE
Admin Menu: update Purchases to Upgrades and add submenu

### PR DESCRIPTION
**Pull request by michaeldcain**
This renames "Purchases" to "Upgrades" in the new unified admin menu. It also adds the "Plans" and "Purchases" submenu items, along with moving "Domains" from "Settings" to "Upgrades".

Screen Shot 2020-12-15 at 10 53 52 AM 

See: https://github.com/Automattic/wp-calypso/issues/48322
Calypso changes: https://github.com/Automattic/wp-calypso/pull/47465/commits/8d37ca1c32d9ade9a6a46b66a0dfc261dd3d6f83

#### Changes proposed in this Pull Request:

* Rename "Purchases" to "Upgrades"
* Add the following sub-pages:
 * "Plans" (/plans/{site_slug})
 * "Domains" (/domains/manage/{site_slug})
 * "Purchases" (/purchases/subscriptions/{site_slug})
 * Remove "Domains" from under "Settings"

#### Jetpack product discussion

pbOQVh-KN-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* `phpunit --testsuite=masterbar`

To test a simple site:
* Sync these changes with your WordPress.com sandbox and sandbox `public-api.wordpress.com`
* Follow the instructions to test the unified nav under "Testing Simple Sites" here: paYJgx-1af-p2
* Visit `/wp-admin/` for a simple site and verify that the Upgrades menu looks like the screenshot above

To test locally:
* Add a mu-plugin that contains: `add_filter( 'jetpack_load_admin_menu_class', '__return_true' );`
* Make sure your site is connected to wp.com and has the masterbar module activated.
* Visit `/wp-admin/` and verify that the Upgrades menu looks like the screenshot above

#### Proposed changelog entry for your changes:

* None needed.

### CODE REVIEW

**Review by jeherve — APPROVED** [review]

**Review by obenland — DISMISSED** [review]
I can't find a single thing wrong with it. 😉

**Review by jeherve — DISMISSED** [review]
Re-approving after merging latest `master` to address unrelated test failures.

File: modules/masterbar/admin-menu/class-admin-menu.php
Comment (michaeldcain) [inline modules/masterbar/admin-menu/class-admin-menu.php:274]: Following the logic from #18041

**Comment by michaeldcain** [comment]
@jeherve: I needed to rebase after #18041, so this might need another approval.

**Review by jeherve — DISMISSED** [review]

**Review by jeherve — APPROVED** [review]

**Pull request merged**
